embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpanImpl.kt

```diff
@@ -23,23 +22,17 @@ import java.util.concurrent.atomic.AtomicReference
 internal class EmbraceSpanImpl(
     private val spanName: String,
     private val openTelemetryClock: Clock,
-    private val spanBuilder: SpanBuilder,
+    private val spanBuilder: EmbraceSpanBuilder,
     override val parent: EmbraceSpan? = null,
-    private val spanRepository: SpanRepository? = null,
-    sessionSpan: Boolean = false
+    private val spanRepository: SpanRepository? = null
 ) : PersistableEmbraceSpan {
 
-    init {
-        if (!sessionSpan) {
-            spanBuilder.updateParent(parent)
-        }
-    }
-
     private val startedSpan: AtomicReference<io.opentelemetry.api.trace.Span?> = AtomicReference(null)
     private var spanStartTimeMs: Long? = null
     private var spanEndTimeMs: Long? = null
     private var status = Span.Status.UNSET
     private val events = ConcurrentLinkedQueue<EmbraceSpanEvent>()
+    private val schemaAttributes = spanBuilder.embraceAttributes.associate { it.toOTelKeyValuePair() }.toMutableMap()
```

Comment:
Store attributes added by the SDK separately so they are not subject to the span attribute limits